### PR TITLE
Add hnt to sol swap

### PR DESCRIPTION
**Pull request by Perronef5**
## CHANGES 📲
- Add hnt to sol swap alert in simulate transaction when simulation fails with insufficient balance
- Add hnt to sol swap bottom sheet on load if user has insufficient balance
- Add Wallet Sign Provider to any other transaction in app that needs to be approved

## TODO
- [x] Figure out why I am getting a network error from the hnt swap url

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -101,40 +103,45 @@ const App = () => {
...
-                              }
-                            />
-                          </>
-                        )}
-                      </WalletConnectProvider>
-                    </AccountProvider>
-                  </LockScreen>
-                )}
-              </OnboardingProvider>
+              <BottomSheetModalProvider>
```
Comment (tyler-whitman) [inline src/App.tsx:106]: Is there a reason we didnt have this in the root before? I remember we moved it out a while back and put it into a bunch of individual files?

File: src/features/account/AccountsScreen.tsx [hunk 2]
```diff
@@ -76,6 +81,46 @@ const AccountsScreen = () => {
...
+        }),
+      })
+
+      if (!decision) return
+
+      await anchorProvider.sendAndConfirm(hntSolConvertTransaction)
+    } catch (e) {
+      logger.error(e)
+    }
+  }, [hntEstimate, serializedTx, t, hasEnoughSol, hntSolConvertTransaction])
```
Comment (tyler-whitman) [inline src/features/account/AccountsScreen.tsx:123]: This is missing `anchorProvider` and `walletSignBottomSheetRef`

File: src/hooks/useHntSolConvert.ts [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        await axios.post(`${baseUrl}/hnt-to-fees`, {
+          wallet: anchorProvider.publicKey.toBase58(),
+        })
+      ).data
+      const tx = Transaction.from(Buffer.from(txRaw))
+      return tx
+    } catch (e) {
+      logger.error(e)
+    }
+  }, [baseUrl])
```
Comment (tyler-whitman) [inline src/hooks/useHntSolConvert.ts:54]: This is missing `anchorProvider`

File: src/hooks/useSimulatedTransaction.ts [hunk 4]
```diff
@@ -130,6 +182,9 @@ export function useSimualtedTransaction(
               'InstructionError":[0,{"Custom":1}]',
             )
           ) {
+            if (!hasEnoughSol) {
+              showHNTConversionAlert()
```
Comment (tyler-whitman) [inline src/hooks/useSimulatedTransaction.ts:186]: This is missing an await, not sure if its needed?

**Review by tyler-whitman — APPROVED** [review]

File: src/App.tsx (hunk 1 above)
Comment (Perronef5) [reply src/App.tsx:106]: Crap your right but I cant remember. Let me just move it back to the individual component

File: src/hooks/useSimulatedTransaction.ts (hunk 4 above)
Comment (Perronef5) [reply src/hooks/useSimulatedTransaction.ts:186]: Hmm I thought I didn't need it here but let me double check

**Pull request merged**